packages/ts/hilla-file-router/src/generateJson.ts

```diff
@@ -0,0 +1,74 @@
+import { readFile } from 'node:fs/promises';
+import { Script } from 'node:vm';
+import ts, { type Node } from 'typescript';
+import type { RouteMeta } from './collectRoutes.js';
+import { prepareConfig, processPattern, type ViewConfig } from './utils.js';
+
+function* traverse(
+  views: RouteMeta,
+  parents: readonly RouteMeta[] = [],
+): Generator<readonly RouteMeta[], undefined, undefined> {
+  const chain = [...parents, views];
+
+  if (views.children.length === 0) {
+    yield chain;
+  }
+
+  for (const child of views.children) {
+    yield* traverse(child, chain);
+  }
+}
+
+type RouteModule = Readonly<{
+  default: unknown;
+  config?: ViewConfig;
+}>;
+
+function* walkAST(node: Node): Generator<Node> {
+  yield node;
+
+  for (const child of node.getChildren()) {
+    yield* walkAST(child);
+  }
+}
+
+export default async function generateJson(views: RouteMeta, configExportName: string): Promise<string> {
```

Comment:
I suggest renaming this function and module to reflect what it exactly does, such `extractViewConfigs`, as `generateJson` is a very common name.